### PR TITLE
ISPN-13337 CLI: expose relay nodes

### PR DESCRIPTION
**Pull request by pruivo**
* Expose cross-site information in JMX
* Renamed "site coordinator" to "relay node"
* Exposed relay nodes in CLI

https://issues.redhat.com/browse/ISPN-13337

### CODE REVIEW

**Comment by tristantarrant** [comment]
LGTM

File: cli/src/main/java/org/infinispan/cli/commands/rest/Site.java [hunk 1]
```diff
@@ -367,4 +369,52 @@ public boolean isHelp() {
...
+            connection.refreshServerInfo();
+            invocation.println(String.valueOf(connection.getRelayNodes()));
+            return CommandResult.SUCCESS;
+         } catch (IOException e) {
+            throw new CommandException(e);
+         }
+      }
+   }
+
+   @CommandDefinition(name = "is-relay-node", description = "Returns true if this node is a relay node.", activator = ConnectionActivator.class)
```
Comment (oraNod) [inline cli/src/main/java/org/infinispan/cli/commands/rest/Site.java:397]:
```suggestion
   @CommandDefinition(name = "is-relay-node", description = "Returns true if the node handles relay messages between clusters.", activator = ConnectionActivator.class)
```

File: cli/src/main/resources/help/site.adoc [hunk 2]
```diff
@@ -87,3 +91,9 @@ Returns the name of the local site. If cross-site replication is not configured,
 
 `site view` +
 Returns a list of names for all sites or an empty list ("[]") if cross-site replication is not configured.
+
+`site is-relay-node` +
+Returns true if this node is a relay node.
```
Comment (oraNod) [inline cli/src/main/resources/help/site.adoc:96]:
```suggestion
Returns true if the node handles RELAY messages between clusters.
```

File: documentation/src/main/asciidoc/topics/ref_rest_cache_managers.adoc [hunk 3]
```diff
@@ -32,8 +32,8 @@ For possible values, check the https://docs.jboss.org/infinispan/9.4/apidocs/org
 * `defined_caches` A list of all caches defined in the cache manager, excluding private caches but including internal caches that are accessible
 * `local_site` The name of the local site. +
 If cross-site replication is not configured, {brandname} returns "local".
-* `site_coordinator` is true if the cache manager is a site coordinator.
-* `site_coordinators_address` is an array of logical addresses of site coordinators.
+* `relay_node` is true if the node is a relay node.
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_cache_managers.adoc:35]:
```suggestion
* `relay_node` is true if the node handles RELAY messages between clusters.
```

File: documentation/src/main/asciidoc/topics/ref_rest_cache_managers.adoc [hunk 4]
```diff
@@ -32,8 +32,8 @@ For possible values, check the https://docs.jboss.org/infinispan/9.4/apidocs/org
...
+* `relay_nodes_address` is an array of logical addresses of relay nodes.
```
Comment (oraNod) [inline documentation/src/main/asciidoc/topics/ref_rest_cache_managers.adoc:36]:
```suggestion
* `relay_nodes_address` is an array of logical addresses for relay nodes.
```

File: cli/src/main/resources/help/site.adoc [hunk 5]
```diff
@@ -87,3 +91,9 @@ Returns the name of the local site. If cross-site replication is not configured,
...
+
+`site relay-nodes` +
+Returns a list of relay nodes (logical names).
```
Comment (oraNod) [inline cli/src/main/resources/help/site.adoc:99]:
```suggestion
Returns a list of relay nodes by their logical names.
```

File: core/src/main/java/org/infinispan/manager/DefaultCacheManager.java [hunk 6]
```diff
@@ -1069,6 +1069,26 @@ public String getClusterName() {
       return cacheManagerInfo.getClusterName();
    }
 
+   @ManagedAttribute(description = "Returns the local site name", displayName = "Local site name", dataType = DataType.TRAIT)
+   public String getSite() {
+      return cacheManagerInfo.getLocalSite();
+   }
+
+   @ManagedAttribute(description = "List of all online sites", displayName = "Online Sites", dataType = DataType.TRAIT)
```
Comment (oraNod) [inline core/src/main/java/org/infinispan/manager/DefaultCacheManager.java:1077]:
```suggestion
   @ManagedAttribute(description = "Lists all online sites", displayName = "Online Sites", dataType = DataType.TRAIT)
```

File: core/src/main/java/org/infinispan/manager/DefaultCacheManager.java [hunk 7]
```diff
@@ -1069,6 +1069,26 @@ public String getClusterName() {
...
+   public String getSiteView() {
+      return String.valueOf(cacheManagerInfo.getSites());
+   }
+
+   @ManagedAttribute(description = "Indicates whether this node is a relay node", displayName = "Is relay node?", dataType = DataType.TRAIT)
+   public boolean isRelayNode() {
+      return cacheManagerInfo.isRelayNode();
+   }
+
+   @ManagedAttribute(description = "List of relay nodes in the local site", displayName = "Relay nodes", dataType = DataType.TRAIT)
```
Comment (oraNod) [inline core/src/main/java/org/infinispan/manager/DefaultCacheManager.java:1087]:
```suggestion
   @ManagedAttribute(description = "Lists relay nodes in the local site", displayName = "Relay nodes", dataType = DataType.TRAIT)
```

**Review by oraNod — APPROVED** [review]
Great work @pruivo Left a few doc suggestions.

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**